src/js/common/schemaform/SaveFormLink.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import LoginModal from '../components/LoginModal';
+
+// TODO: Come up with a better name than SaveFormLink
+class SaveFormLink extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      modalOpened: false
+    };
+  }
+
+  openLoginModal = () => {
+    this.setState({ modalOpened: true });
+  }
+
+  closeLoginModal = () => {
+    this.setState({ modalOpened: false });
+  }
+
+  render() {
+    const {
+      saveForm,
+      savedStatus
+    } = this.props;
+    let content = (<div>
+      <a onClick={this.openLoginModal}>Sign in before saving your application</a>
+      <LoginModal
+          key={1}
+          title="Sign in to save your application"
+          onClose={this.closeLoginModal}
+          visible={this.state.modalOpened}/>
+    </div>);
+
+    if (this.props.loggedIn) {
+      content = <a onClick={saveForm}>Save and come back later</a>;
+    }
+
+    if (savedStatus === 'no-auth') {
+      content = <p>no-auth message</p>;
+    } else if (savedStatus === 'failure') {
+      content = <p>failure message</p>;
+    } else if (savedStatus === 'pending') {
+      content = <p>spinner or something</p>;
+    } else if (savedStatus === 'success') {
+      content = <p>success message</p>;
+      // TODO: Redirect to a page like: https://marvelapp.com/2hj59b1/screen/28358414
+    }
+
+    // TODO: If we get a no-auth, we should reset the link after login
+    //  Or, as a temporary solution, include the save link in the no-auth
+    //  message..?
+    return content;
+  }
+}
+
+SaveFormLink.propTypes = {
+  saveForm: PropTypes.func.isRequired,
+  savedStatus: PropTypes.string.isRequired
+};
+
+// I'd rather not have this linked to redux, but then again, I'd rather not pass
```

Comment:
It seems like this is generally going to be the direct child of a connected component, so there shouldn't be much prop threading. And I'm not sure we should worry about re-renders if this is going to route to another page